[Bug]: Dependency on sharp in @storybook/nextjs should be optional

### Describe the bug

As of [7.6.0](https://github.com/storybookjs/storybook/compare/v7.5.3...v7.6.0) @nextjs/storybook has a direct dependency on [sharp](https://sharp.pixelplumbing.com/install), in order to handle AVIF processing. This is an optional dependency in nextjs itself so presents an issue for anyone (like us) who are running Storybook inside of a tightly constrainted CI environment - we have no need for AVIF or Next.js's image optimisation, so we don't need the sharp dependency and don't want to whitelist the download domains in order for it to download binaries at install time.

It would be better for this to be an optional dependency, and to throw an error if an AVIF file is detected - this would match Next's own behaviour (https://nextjs.org/docs/messages/install-sharp)

### To Reproduce

_No response_

### System

```bash
Storybook Environment Info:

  System:
    OS: macOS 14.1.1
    CPU: (10) arm64 Apple M1 Max
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.10.0 - ~/.asdf/installs/nodejs/20.10.0/bin/node
    Yarn: 1.22.19 - ~/.asdf/shims/yarn
    npm: 10.2.4 - ~/.asdf/plugins/nodejs/shims/npm <----- active
  Browsers:
    Chrome: 119.0.6045.199
    Edge: 119.0.2151.97
    Safari: 17.1
  npmPackages:
    @storybook/addon-actions: 7.6.3 => 7.6.3
    @storybook/addon-essentials: 7.6.3 => 7.6.3
    @storybook/addon-interactions: 7.6.3 => 7.6.3
    @storybook/addon-links: 7.6.3 => 7.6.3
    @storybook/blocks: 7.6.3 => 7.6.3
    @storybook/nextjs: 7.6.3 => 7.6.3
    @storybook/react: 7.6.3 => 7.6.3
    @storybook/testing-library: 0.2.2 => 0.2.2
    chromatic: 9.1.0 => 9.1.0
    eslint-plugin-storybook: 0.6.15 => 0.6.15
    storybook: 7.6.3 => 7.6.3
```


### Additional context

_No response_

@LifeAsDev I'm having the exact same problem.  `npx storybook@latest init` fails to install.  `npm ERR! sharp: Installation error: unable to get local issuer certificate`, which is why I'm assuming @matmannion mentions whitelisting.

--- a/code/frameworks/nextjs/package.json
+++ b/code/frameworks/nextjs/package.json
@@ -114,7 +114,6 @@
     "resolve-url-loader": "^5.0.0",
     "sass-loader": "^12.4.0",
     "semver": "^7.3.5",
-    "sharp": "^0.32.6",
     "style-loader": "^3.3.1",
     "styled-jsx": "5.1.1",
     "ts-dedent": "^2.0.0",
@@ -131,6 +130,9 @@
     "next": "^14.1.0",
     "typescript": "^5.3.2",
     "webpack": "^5.65.0"
+  },
+  "optionalDependencies": {
+    "sharp": "^0.33.3"
   },
   "peerDependencies": {
     "next": "^13.5.0 || ^14.0.0",

--- a/code/frameworks/nextjs/src/next-image-loader-stub.ts
+++ b/code/frameworks/nextjs/src/next-image-loader-stub.ts
@@ -3,6 +3,7 @@ import imageSizeOf from 'image-size';
 import type { RawLoaderDefinition } from 'webpack';
 import type { NextConfig } from 'next';
 import { cpus } from 'os';
+import { NextJsSharpError } from '@storybook/core-events/preview-errors';
 
 interface LoaderOptions {
   filename: string;
@@ -51,9 +52,7 @@ const nextImageLoaderStub: RawLoaderDefinition<LoaderOptions> = async function N
       width = result.width;
       height = result.height;
     } else {
-      throw new Error(
-        'You have to install sharp in order to use image optimization features in Next.js.'
-      );
+      throw new NextJsSharpError();
     }
   } else {
     const result = imageSizeOf(this.resourcePath);

--- a/code/frameworks/nextjs/src/next-image-loader-stub.ts
+++ b/code/frameworks/nextjs/src/next-image-loader-stub.ts
@@ -2,7 +2,6 @@ import { interpolateName } from 'loader-utils';
 import imageSizeOf from 'image-size';
 import type { RawLoaderDefinition } from 'webpack';
 import type { NextConfig } from 'next';
-import sharp from 'sharp';
 import { cpus } from 'os';
 
 interface LoaderOptions {
@@ -10,10 +9,19 @@ interface LoaderOptions {
   nextConfig: NextConfig;
 }
 
-if (sharp.concurrency() > 1) {
-  // Reducing concurrency reduces the memory usage too.
-  const divisor = process.env.NODE_ENV === 'development' ? 4 : 2;
-  sharp.concurrency(Math.floor(Math.max(cpus().length / divisor, 1)));
+let sharp: typeof import('sharp') | undefined;
+
+try {
+  sharp = require('sharp');
+  if (sharp && sharp.concurrency() > 1) {
+    // Reducing concurrency reduces the memory usage too.
+    const divisor = process.env.NODE_ENV === 'development' ? 4 : 2;
+    sharp.concurrency(Math.floor(Math.max(cpus().length / divisor, 1)));
+  }
+} catch (e) {
+  console.warn(
+    'You have to install sharp in order to use image optimization features in Next.js. AVIF support is also disabled.'
+  );
 }
 
 const nextImageLoaderStub: RawLoaderDefinition<LoaderOptions> = async function NextImageLoader(
@@ -37,10 +45,16 @@ const nextImageLoaderStub: RawLoaderDefinition<LoaderOptions> = async function N
   let height;
 
   if (extension === 'avif') {
-    const transformer = sharp(content);
-    const result = await transformer.metadata();
-    width = result.width;
-    height = result.height;
+    if (sharp) {
+      const transformer = sharp(content);
+      const result = await transformer.metadata();
+      width = result.width;
+      height = result.height;
+    } else {
+      throw new Error(
+        'You have to install sharp in order to use image optimization features in Next.js.'
+      );
+    }
   } else {
     const result = imageSizeOf(this.resourcePath);
     width = result.width;

--- a/code/lib/core-events/src/errors/preview-errors.ts
+++ b/code/lib/core-events/src/errors/preview-errors.ts
@@ -27,6 +27,7 @@ export enum Category {
   RENDERER_VUE = 'RENDERER_VUE',
   RENDERER_VUE3 = 'RENDERER_VUE3',
   RENDERER_WEB_COMPONENTS = 'RENDERER_WEB-COMPONENTS',
+  FRAMEWORK_NEXTJS = 'FRAMEWORK_NEXTJS',
 }
 
 export class MissingStoryAfterHmrError extends StorybookError {
@@ -233,5 +234,21 @@ export class StoryStoreAccessedBeforeInitializationError extends StorybookError 
 
     It is not recommended to use methods directly on the Story Store anyway, in Storybook 9 we will
     remove access to the store entirely`;
+  }
+}
+
+export class NextJsSharpError extends StorybookError {
+  readonly category = Category.FRAMEWORK_NEXTJS;
+
+  readonly code = 1;
+
+  readonly documentation = 'https://storybook.js.org/docs/get-started/nextjs#faq';
+
+  template() {
+    return dedent`
+    You are importing avif images, but you don't have sharp installed.
+
+    You have to install sharp in order to use image optimization features in Next.js.
+    `;
   }
 }

--- a/docs/get-started/nextjs.md
+++ b/docs/get-started/nextjs.md
@@ -865,6 +865,24 @@ You might get this if you're using Yarn v2 or v3. See [Notes for Yarn v2 and v3 
 
 The `@storybook/nextjs` package abstracts the Webpack 5 builder and provides all the necessary Webpack configuration needed (and used internally) by Next.js. Webpack is currently the official builder in Next.js, and Next.js does not support Vite, therefore it is not possible to use Vite with `@storybook/nextjs`. You can use `@storybook/react-vite` framework instead, but at the cost of having a degraded experience, and we won't be able to provide you official support.
 
+### Error: You are importing avif images, but you don't have sharp installed. You have to install sharp in order to use image optimization features in Next.js.
+
+`sharp` is a dependency of Next.js's image optimization feature. If you see this error, you need to install `sharp` in your project.
+
+```bash
+npm install sharp
+```
+
+```bash
+yarn add sharp
+```
+
+```bash
+pnpm add sharp
+```
+
+You can refer to the [Install `sharp` to Use Built-In Image Optimization](https://nextjs.org/docs/messages/install-sharp) in the Next.js documentation for more information.
+
 ## API
 
 ### Parameters

